streams/src/main/java/org/apache/kafka/streams/processor/internals/StandbyTask.java

```diff
@@ -247,6 +247,9 @@ private void close(final boolean clean) {
                 "state manager close",
                 log
             );
+        } else if (state() == State.CLOSED) {
+            log.trace("Skip closing since state is {}", state());
+            return;
         } else {
             throw new IllegalStateException("Illegal state " + state() + " while closing standby task " + id);
```

Comment:
`Illegal state` -> `Unknown state`?

Reply:
The state could be `RESTORING` what is _illegal_ but not _unknown_ -- We would need more conditions to distinguish both cases (introducing `switch()` would be helpful for this case). Thoughts?

Reply:
I see. I guess personally I'd prefer to use a `switch` and just case all states explicitly, to make it clear when an action is a no-op for a certain state

Reply:
I'd be in favor... I can see how the choice can go both ways on if/else vs. switch, but at the end of the day, it's nice just to do the same thing everywhere. I think switch tends to be nicer when you need to enumerate most/all of the cases, which seems to be the direction we're heading here.

Reply:
I update most. Only left a few for which we only check a single state (seems overkill to use `switch()` for this case). Is that ok with you?